node/src/components/rpc_server/rpcs/state.rs

```diff
@@ -1273,14 +1168,9 @@ pub(super) async fn get_state_root_hash_and_optional_header<REv: ReactorEventT>(
                 .await
             {
                 None => {
-                    let error_msg = format!(
-                        "{} failed to retrieve block header at height {}",
-                        method_name, block_height
-                    );
-                    Err(warp_json_rpc::Error::custom(
-                        ErrorCode::NoSuchBlock as i64,
-                        error_msg,
-                    ))
```

Comment:
I haven't reviewed every single line in this file super-closely but it all seems to be very mechanical work adapting the old API to a new one.

Reply:
Yes, there should be no externally-observable changes to the actual logic inside the specific RPC handlers.